saleor/webhook/validators.py

```diff
@@ -0,0 +1,46 @@
+import re
+from typing import Dict
+
+from django.core.exceptions import ValidationError
+
+HEADERS_NUMBER_LIMIT = 5
+HEADERS_LENGTH_LIMIT = 78
+
+
+def custom_headers_validator(headers) -> Dict[str, str]:
+    """Validate headers in accordance with RFC5322.
+
+    https://www.rfc-editor.org/rfc/rfc5322#section-2.2
+    """
+    if len(headers) > HEADERS_NUMBER_LIMIT:
+        raise ValidationError(
+            f"Number of headers exceeds the limit: {HEADERS_NUMBER_LIMIT}."
+        )
+
+    key_chars_allowed = "".join([chr(i) for i in range(33, 127) if i != ord(":")])
+    value_chars_allowed = key_chars_allowed + chr(32) + chr(9)
+
+    for key, value in headers.items():
+        try:
+            header = ": ".join([key, value])
+        except TypeError:
+            raise ValidationError("One of the header can't be converted to string.")
```

Comment:
Can't we try to add here `key`?  I think, that it would be helpful in case of understanding what is wrong

Reply:
If it can't be converted to string, how can I add it to string message?

Reply:
So in which scenario, the `key` cannot be converted to string? 
You're getting headers as a JSONString from API, and from manifest  (probably also as a json or jsonstring). So in both scenarios, the key & value need to be serializable

The current implementation will fail, when you will have numeric value for `key` or `value` will be for example a `dict` or `list`, but I don't know which case doesn't allow us to provide a `key` here.